File: src/main/java/com/bazbatlabs/smashballs/controllers/Controller.java
package com.bazbatlabs.smashballs.controllers;

import javax.microedition.khronos.opengles.GL10;
import android.view.KeyEvent;

public interface Controller {
    public void draw(int screenWidth, int screenHeight);
    public Controller update();
    public boolean onKeyDown(int keyCode, KeyEvent event);
    public boolean onKeyUp(int keyCode, KeyEvent event);
    public void changeSurface(int width, int height);
}

File: src/main/scala/lib/Controller.scala
package com.bazbatlabs.bricksmash.lib

import javax.microedition.khronos.opengles.GL10
import android.view.KeyEvent

trait Controller {
  def draw(screenWidth: Int, screenHeight: Int)
  def update(): Controller
  def onKeyDown(keyCode: Int, event: KeyEvent): Boolean
  def onKeyUp(keyCode: Int, event: KeyEvent): Boolean
  def changeSurface(width: Int, height: Int)
}
